### DISCUSSION TITLE
Tail Sampling on Opentelemetry Sidecar Collector

### DISCUSSION THREAD

**Discussion by Sayyadd** [open-telemetry/opentelemetry-collector#10038] [Q&A] [answered]
Hi,
I couldn't find accurate answer on documentation, that's why I am asking here.
Question is:  If I deploy opentelemetry collector in sidecar mode, is it possible to apply tail sampling on that collector or I should do it on gateway collector ?

**Comment by yurishkuro** [comment, marked as answer]
Tail sampling needs to collect all spans for a trace before making a decision. A sidecar collector would normally only see a subset of a trace, assuming you have a distributed system with many services and many sidecars. So yes, you need to do tail sampling in the last stage collector which accepts traffic already partitioned by trace IDs.

**Comment by Sayyadd** [comment]
Thank you @yurishkuro
